applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_solver_fractionalstep.py

```diff
@@ -128,63 +114,99 @@ def AddDofs(self):
-                                                                                self.settings["reform_dofs_at_each_step"].GetBool())
-
-        self.solver_settings.SetEchoLevel(self.settings["echo_level"].GetInt())
-
-        self.solver_settings.SetStrategy(TrilinosFluid.TrilinosStrategyLabel.Velocity,
-                                         self.velocity_linear_solver,
-                                         self.settings["velocity_tolerance"].GetDouble(),
-                                         self.settings["maximum_velocity_iterations"].GetInt())
-
-        self.solver_settings.SetStrategy(TrilinosFluid.TrilinosStrategyLabel.Pressure,
-                                         self.pressure_linear_solver,
-                                         self.settings["pressure_tolerance"].GetDouble(),
-                                         self.settings["maximum_pressure_iterations"].GetInt())
-
-        self.solver = TrilinosFluid.TrilinosFSStrategy(self.computing_model_part,
-                                                       self.solver_settings,
-                                                       self.settings["predictor_corrector"].GetBool(),
-                                                       KratosFluid.PATCH_INDEX)
-
-        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, self.settings["dynamic_tau"].GetDouble())
-        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.OSS_SWITCH, self.settings["oss_switch"].GetInt())
-
-        (self.solver).Initialize()
-
-        KratosMultiphysics.Logger.PrintInfo("TrilinosNavierStokesSolverFractionalStep","Initialization TrilinosNavierStokesSolverFractionalStep finished")
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
 
     def Finalize(self):
-        self.solver.Clear()
+        self.GetSolutionStrategy().Clear()
+
+    def get_epetra_communicator(self):
+        if not hasattr(self, '_epetra_communicator'):
+            self._epetra_communicator = KratosTrilinos.CreateCommunicator()
+        return self._epetra_communicator
+
+    def _CreateScheme(self):
+        pass
+
+    def _CreateLinearSolver(self):
+        # Create the pressure linear solver
+        pressure_linear_solver_configuration = self.settings["pressure_linear_solver_settings"]
+        pressure_linear_solver = trilinos_linear_solver_factory.ConstructSolver(pressure_linear_solver_configuration)
+        # Create the velocity linear solver
+        velocity_linear_solver_configuration = self.settings["velocity_linear_solver_settings"]
+        velocity_linear_solver = trilinos_linear_solver_factory.ConstructSolver(velocity_linear_solver_configuration)
+        # Return a tuple containing both linear solvers
+        return (pressure_linear_solver, velocity_linear_solver)
+
+    def _CreateConvergenceCriterion(self):
+        pass
+
+    def _CreateBuilderAndSolver(self):
+        pass
```

Comment:
this seems wrong...

Reply:
These are not required in the FS solver. We can remove them if you prefer it but this would imply that the FS strategy would eventually use the implementation of the base class (although it is never called).

Reply:
how abt throwing in those functions then?
or will this not work bcs they are still called somewhere in the workflow?

Reply:
They are not used in the FS solver so, from a practical point of view, there will be no difference in doing a pass or throwing an error. However I found more consistent to implement a "do nothing" in this case.